nifi-nar-bundles/nifi-site-to-site-reporting-bundle/nifi-site-to-site-reporting-task/src/main/java/org/apache/nifi/reporting/SiteToSiteProvenanceReportingTask.java

```diff
@@ -174,34 +177,47 @@ public void onUnscheduled() {
         return properties;
     }
 
-    private Map<String,String> createComponentMap(final ProcessGroupStatus status) {
-        final Map<String,String> componentMap = new HashMap<>();
+    private ComponentMapHolder createComponentMap(final ProcessGroupStatus status) {
```

Comment:
This method probably should be in ProvenanceEventConsumer instead of each ReportingTask implementation so that other ReportingTasks can get benefits from it. How do you think?

Reply:
I agree, will move it there.